chains/substrate/client/client.go

```diff
@@ -75,15 +77,37 @@ func (c *SubstrateClient) Transact(method string, args ...interface{}) (*types.H
 		Tip:                types.NewUCompactFromUInt(c.tip),
 		TransactionVersion: rv.TransactionVersion,
 	}
-	h, err := c.signAndSendTransaction(o, &ext)
+	sub, err := c.submitAndWatchExtrinsic(o, &ext)
 	if err != nil {
-		return nil, fmt.Errorf("submission of extrinsic failed: %w", err)
+		return "", nil, fmt.Errorf("submission of extrinsic failed: %w", err)
 	}
 
-	log.Debug().Msgf("Extinsic call succededed... method %s, sender %s, nonce %d", method, c.key.Address, nonce)
+	hash, err := substrate.ExtrinsicHash(ext)
+	if err != nil {
+		return "", nil, err
+	}
+
+	log.Info().Str("extrinsic", hash).Msgf("Extrinsic call submitted... method %s, sender %s, nonce %d", method, c.key.Address, nonce)
 	c.nonce = nonce + 1
 
-	return &h, nil
+	return hash, sub, nil
+}
+
+func (c *SubstrateClient) TrackExtrinsic(extHash string, sub *author.ExtrinsicStatusSubscription, errChn chan error) {
+	defer sub.Unsubscribe()
+	subChan := sub.Chan()
+	for {
+		status := <-subChan
+		if status.IsInBlock {
```

Comment:
Is it possible that the transaction never got finalized, and we end up in infinity loop?

Reply:
I guess it might be possible. I'll create a context timeout just to be safe.